portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function getUserName(){
+    const userName =  fetch("/data").then(response => response.text()).then(name);
+    return userName;
```

Comment:
optional nit: you can one-line this and just do
return fetch('/data").then(.....